spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/source/SparkCopyOnWriteScan.java

```diff
@@ -102,6 +103,14 @@ public NamedReference[] filterAttributes() {
 
   @Override
   public void filter(Filter[] filters) {
+    Preconditions.checkState(
+        Objects.equals(snapshotId(), currentSnapshotId()),
+        "Runtime file filtering is not possible: the table has been concurrently refreshed. "
```

Comment:
I initially thought about not doing runtime filtering if the table has been refreshed to avoid failing the query. However, we change such a critical place. I think it is better to let the use know rather than proceeding with the query. Otherwise, the jobs that leverage multiple threads may take much more resources.